pkg/frontend/mysql_cmd_executor.go

```diff
@@ -3931,12 +3959,15 @@ func (mce *MysqlCmdExecutor) doComQuery(requestCtx context.Context, sql string)
 				if st.Local {
 					loadLocalErrGroup = new(errgroup.Group)
 					loadLocalErrGroup.Go(func() error {
-						return mce.processLoadLocal(proc.Ctx, st.Param, loadLocalWriter)
+						return mce.processLoadLocal(proc.Ctx, st.Param, loadLocalWriter, loadLocalEarlyStop)
 					})
 				}
 			}
 
 			if err = runner.Run(0); err != nil {
+				if loadLocalEarlyStop != nil {
+					loadLocalEarlyStop.Store(true)
```

Comment:
Put 
```
....
loadLocalErrGroup.Wait();
....
```
in an independent function.
When the `loadLocalEarlyStop` is set true, the function should called.